pkg/manifest/subscription.go

```diff
@@ -116,12 +116,29 @@ func subscriptionService(subscriptionOptions subscription.ImageOptions, serviceO
 
 	var commands []string
 	if subscriptionOptions.Rhc {
+		var curlToAssociateSystem string
 		// Use rhc for registration instead of subscription manager
-		commands = []string{fmt.Sprintf("/usr/bin/rhc connect --organization=${ORG_ID} --activation-key=${ACTIVATION_KEY} --server %s", subscriptionOptions.ServerUrl)}
+		rhcConnect := fmt.Sprintf("/usr/bin/rhc connect --organization=${ORG_ID} --activation-key=${ACTIVATION_KEY} --server %s", subscriptionOptions.ServerUrl)
+		if subscriptionOptions.TemplateUUID != "" {
+			patchURL := subscriptionOptions.PatchURL
+			if patchURL[len(patchURL)-1] == '/' {
+				patchURL = patchURL[:len(patchURL)-1]
+			}
+			addTemplateURL := fmt.Sprintf("%s/templates/%s/subscribed-systems", patchURL, subscriptionOptions.TemplateUUID)
+			curlToAssociateSystem = fmt.Sprintf("curl -v --retry --proxy %s %s --cert /etc/pki/consumer/cert.pem  --key /etc/pki/consumer/key.pem -X PATCH", subscriptionOptions.Proxy, addTemplateURL)
+		} else if subscriptionOptions.TemplateName != "" {
+			rhcConnect += fmt.Sprintf(" --content-template %s", subscriptionOptions.TemplateName)
+		}
+		commands = append(commands, rhcConnect)
 		// insights-client creates the .gnupg directory during boot process, and is labeled incorrectly
 		commands = append(commands, "restorecon -R /root/.gnupg")
 		// execute the rhc post install script as the selinuxenabled check doesn't work in the buildroot container
 		commands = append(commands, "/usr/sbin/semanage permissive --add rhcd_t")
+		// register to template if template uuid is specified
+		if curlToAssociateSystem != "" {
+			commands = append(commands, curlToAssociateSystem)
+			commands = append(commands, "/usr/sbin/subscription-manager refresh")
```

Comment:
Please move these to the `if subscriptionOptions.TemplateUUID != "" ` statement block, which is the only case when `curlToAssociateSystem` is actually set. You have it done this way in the case when `rhc` is not used.

Reply:
I agree it feels  a little messy, but I did it this way to ensure the curl command runs after `rhc connect`. The template needs to be associated to the system with the patch API after the system is registered.

Reply:
Thanks for the additional context.

If `TemplateUUID` and `TemplateName` are mutually exclusive, as they are in your implementation, the code is fine from my PoV.